pkg/ipmi/rakp_message_1_test.go

```diff
@@ -74,5 +74,30 @@ func TestRAKPMessage1SerializeTo(t *testing.T) {
 		if !bytes.Equal(got, test.wire) {
 			t.Errorf("serialize %v = %v, want %v", test.layer, got, test.wire)
 		}
+
+		var decoded RAKPMessage1
+		if err := decoded.DecodeFromBytes(got, nil); err != nil {
+			t.Errorf("decode %v = error %v", test.layer, err)
+			continue
+		}
+		r := test.layer
+		if r.Tag != decoded.Tag {
```

Comment:
Nit: we want to check whether `decoded.Tag` (the thing that might be incorrect) is equal to `r.Tag` (the reference value), rather than the other way around, so we'd typically swap the operands, i.e. `decoded.Tag != r.Tag`. As written, this is sometimes called a [Yoda condition](https://en.wikipedia.org/wiki/Yoda_conditions).

To ease writing an if-statement for every field, which is easy to get wrong, I originally used `cmp.Diff()` for comparing the entire produced and reference structs - see `rakp_message_2_test.go`.

Reply:
Thank for the hint to `cmp.Diff`: it makes the test much shorter and any failure much more readable!

I had to add `test.layer.BaseLayer.Contents = got`, but everything seems now correct.